node/src/components/small_network.rs

```diff
@@ -563,8 +625,32 @@ where
 
     /// Returns the set of connected nodes.
     #[cfg(test)]
-    pub(crate) fn connected_nodes(&self) -> HashSet<NodeId> {
-        self.outgoing.keys().cloned().collect()
+    pub(crate) fn outgoing_connected_nodes(&self) -> HashSet<NodeId> {
+        self.outgoing
+            .iter()
+            .filter_map(|(n, o)| {
```

Comment:
I strongly dislike single-char variable names.  We have a tendency here to lean more towards longer, explicit variable naming than brevity.